Add LanceDB vectorstore

Pixeladed opened the pull request:
[LanceDB](https://github.com/lancedb/lancedb) is a vector database based on [Lance](https://github.com/eto-ai/lance) columnar data format.

This PR adds basic integration with lancedb as a vectorstore

Commit:
add lancedb as optional

Commit:
impl add texts

Commit:
accept ids or generate new ids

Commit:
add similarity search

Commit:
add from texts

Commit:
remove duplicate checks

Commit:
accept table instead of conn

Commit:
add example

Commit:
fix list key

Commit:
Merge branch 'hwchase17:master' into master

Commit:
add lancedb to pkg

hwchase17 reviewed (commented):
this is great! can we add an example notebook to the docs?

Commit:
add simple test, strict type embedding func

Commit:
add tests

Commit:
fix signature mismatch

Commit:
add lancedb example notebook

Commit:
add an ecosystem.md

Commit:
Merge remote-tracking branch 'upstream-origin/master'

Commit:
resolve conflict in pyproj

Pixeladed commented:
Updated:
- Fix signature mismatch with base `VectorStore`
- use `Embeddings` instead of `Callable` embedding function
- Added some simple tests
- Added an md file in ecosystem and a notebook example - I'm using `FakeEmbeddings` here, not sure if that's appropriate
